src/atopile/config.py

```diff
@@ -1,387 +1,309 @@
-        del data[
-            "location"
-        ]  # The location is saved by the literal location of the file
-        return data
 
+class GlobalConfigSettingsSource(ConfigFileSettingsSource):
     @classmethod
-    def structure(cls, data: dict) -> "ProjectConfig":
-        """Make a config object from a dictionary."""
-        try:
-            return _converter.structure(cls._sanitise_dict_keys(data), cls)
-        except* KeyError as exs:
-            for ex in exs.exceptions:
-                # FIXME: make this less shit
-                raise AtoConfigError(f"Bad key in config {repr(ex)}") from ex
-        raise ValueError("Failed to structure config")
-
-    def patch_config(self, original: dict) -> dict:
-        """Apply a delta between the original and the current config."""
-
-        # delayed import to improve startup time
-        # because deepdiff loads pandas
-        from deepdiff import DeepDiff, Delta
-
-        original_cfg = self.structure(original)
-
-        # Here we need to work around some structural changes
-        # FIXME: the ideal behaviour here would be to default back to
-        # the new structure whenever there's a conflict, but I can't
-        # find a hook to callback to a "conflict-resolver" or the likes
-        # and the exceptions don't have sufficient information to easily find them
-        original_deps_by_name = {d.name: d for d in original_cfg.dependencies}
-        original_dep_indicies = {
-            d.name: i for i, d in enumerate(original_cfg.dependencies)
-        }
-        for d in self.dependencies:
-            if d.name in original_deps_by_name and d != original_deps_by_name[d.name]:
-                del original["dependencies"][original_dep_indicies[d.name]]
-        original_cfg = self.structure(original)
-        # Kill me... I'm sorry
-
-        diff = DeepDiff(
-            self._unsanitise_dict_keys(_converter.unstructure(original_cfg)),
-            self._unsanitise_dict_keys(_converter.unstructure(self)),
-        )
-
-        delta = Delta(diff)
-        return cast_assert(dict, original + delta)
-
-    def save_changes(self, location: Optional[Path] = None) -> None:
+    def find_config_file(cls) -> Path | None:
         """
-        Save the changes to the config object
+        Find the global config file in the user's home directory.
         """
-        if location is None:
-            location = self.location / CONFIG_FILENAME
+        match platform.system():
+            case "Darwin" | "Linux":
+                config_dir = (
+                    Path(os.environ.get("XDG_CONFIG_HOME", Path.home() / ".config"))
+                    / APPLICATION_NAME
+                )
+            case "Windows":
+                # TODO: @windows
+                raise UserException(f"Unsupported platform: {platform.system()}")
+            case _:
+                raise UserException(f"Unsupported platform: {platform.system()}")
```

Comment:
I'd typically advocate for the simplest solution, and use it even on OSs that less often follow this pattern. Even `ssh` does under Windows, if I remember correctly. Which means there's precedence for forcing it

```suggestion
        config_dir = Path.home() / ".config" / APPLICATION_NAME
```